### ISSUE TITLE
Unable on how to set region_size for tablets

### ISSUE THREAD

**Issue by ghost** [hyprwm/Hyprland#3541]
### Hyprland Version

0.30.0

### Bug or Regression?

Bug

### Description

Ever since the new tablet configuration features were added https://wiki.hyprland.org/Configuring/Variables/#tablet I wanted to use the region_size in my config file and define it. However, I am unsure on how to set a value that is of the vec2 type.

However, none of them work and gives me an error of error setting value <0 0> for field <input:tablet:region_size>: No such field.

### How to reproduce

Try setting vec2 values for region_size

I have tried:

region_size = 0 0
region_size = 0, 0
region_size = [0 0]
region_size  = [0, 0]
same thing with ()

**Comment by pdamianik** [comment]
The error you are getting gets produced, when you try to set a device specific configuration value and either the matching device can't be found or the configuration key is invalid for a device configuration. I suspect it is the latter in your case, as device specific keys are just the last segment of the key (`region_size` for your configuration). The configuration for this value for a specific device would be
```
device:[name] {
    region_size = [value]
}
```

If you want to set this value globally for all tablets you can use the `input:tablet:region_size` key:

```
input {
    tablet {
        region_size: [value]
    }
}
```

**Comment by notmentaloutlaw** [comment]
Currently I have this

```
input {
    tablet {
        output = DP-1
        region_size = 1.0, 0.9
    }
}

```
Are you suggesting that instead of = I use a : ?

**Comment by pdamianik** [comment]
No, sorry, that was my mistake. `=` is the expected assignment operator. Also I don't use a tablet, I just extrapolate from what I read in the documentation, the source code and your errors. According to [ConfigManager.cpp:432-437](https://github.com/hyprwm/Hyprland/blob/v0.30.0/src/config/ConfigManager.cpp#L432-L437) a vector gets parsed as two floats separated by a space. But if you had any issue with the vector parsing you would get the error at [ConfigManager.cpp:440](https://github.com/hyprwm/Hyprland/blob/v0.30.0/src/config/ConfigManager.cpp#L440) namely without the "No such field" message you are getting. That particular error message can only be found at [ConfigManager.cpp:359](https://github.com/hyprwm/Hyprland/blob/v0.30.0/src/config/ConfigManager.cpp#L359) and [ConfigManager.cpp:383](https://github.com/hyprwm/Hyprland/blob/v0.30.0/src/config/ConfigManager.cpp#L383). These two cases happen when either the device name could not be matched to a device (configuration) or the configuration for the device does not contain the specified key. So that's why I assumed that you had an issue related to a device specific configuration, but the config snippet you provided appears to be fine, so it seems to be something else in your config. Your complete configuration will be needed for further support.

**Comment by pdamianik** [comment]
Nevermind the hole `input:tablet:` section has only been added on the master branch. The wiki defaults to the master branch, but you can select a specific version in the menu. The section would be right below [`input:touchdevice:`](https://wiki.hyprland.org/0.30.0/Configuring/Variables/#touchdevice) ([`input:tablet:`](https://wiki.hyprland.org/Configuring/Variables/#tablet))

**Comment by engibengi** [comment]
I think I figured it out:
when encountering the same problem I happened to notice that closing the touchdevice section also happens to close the entire input section
e.g.:
```
input {
       touchdevice {
                    transform = 1
       }
       follow_mouse = 2
}
```
@notmentaloutlaw I think you can solve your problem setting the touchdevice subsection after the tablet one if you didn't try that yet:
```
input {
       ...
       tablet {
               ...
       }
       touchdevice {
                    ...
       }
}
```
My region size is now working
Bear in mind that I got an error similar to yours, but not equal:
`error setting value <0 0> for field tablet:region_size: No such field.`
(the `input:` before `tablet:` is missing)
So it may not be the solution you were looking for.
Anyway, I opened an issue #3992.
I'm tagging it in case that solves this problem too!

**Comment by asiantuntija** [comment]
This worked for me but I had to declare both, region_position and region_size, otherwise nothing was happening. I would argue that setting just one value should be enough:

- if region_position is set then region_size should be bottom_right_pixel_position - region_position, some special monitor setups might have problems with this though
- if region_size is set then position should be just 0,0 by default

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
